test/common/stats/allocator_impl_test.cc

```diff
@@ -524,6 +524,58 @@ TEST_F(AllocatorImplTest, ForEachSinkedGauge) {
   EXPECT_EQ(num_iterations, 0);
 }
 
+TEST_F(AllocatorImplTest, ForEachSinkedGaugeHidden) {
+  GaugeSharedPtr sinked_gauge;
+  GaugeSharedPtr unsinked_gauge;
+
+  auto unhidden_stat_name = makeStat(absl::StrCat("unhidden.gauge"));
+  auto hidden_stat_name = makeStat(absl::StrCat("hidden.gauge"));
+
+  size_t num_gauges = 0;
+  size_t num_iterations = 0;
+
+  sinked_gauge =
+      alloc_.makeGauge(unhidden_stat_name, StatName(), {}, Gauge::ImportMode::Accumulate);
+
+  unsinked_gauge =
+      alloc_.makeGauge(hidden_stat_name, StatName(), {}, Gauge::ImportMode::HiddenAccumulate);
+
+  alloc_.forEachSinkedGauge([&num_gauges](std::size_t size) { num_gauges = size; },
+                            [&num_iterations](Gauge&) { ++num_iterations; });
```

Comment:
rather than just checking num_iterations can you actually make sure that the only call to the lambda is the sinked gauge?

Reply:
Done